pyvista/plotting/plotting.py

```diff
@@ -2599,30 +2609,43 @@ def get_image_depth(self,
 
         Parameters
         ----------
-        fill_value : float
-            Fill value for points in image that don't include objects in scene.
-            To not use a fill value, pass ``None``.
+        fill_value : float, optional
+            Fill value for points in image that do not include objects
+            in scene.  To not use a fill value, pass ``None``.
 
-        reset_camera_clipping_range : bool
-            Reset the camera clipping range to include data in view?
+        reset_camera_clipping_range : bool, optional
+            Reset the camera clipping range to include data in view.
 
         Returns
         -------
         image_depth : numpy.ndarray
-            Image of depth values from camera orthogonal to image plane
+            Image of depth values from camera orthogonal to image
+            plane.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> plotter = pyvista.Plotter()
+        >>> _ = plotter.add_mesh(pyvista.Sphere())
+        >>> plotter.show()  # doctest:+SKIP
+        >>> zval = plotter.get_image_depth()  # doctest:+SKIP
 
         Notes
         -----
         Values in image_depth are negative to adhere to a
         right-handed coordinate system.
 
         """
-        if not hasattr(self, 'ren_win') and hasattr(self, 'last_image_depth'):
+        # allow no render window
+        if not hasattr(self, 'ren_win') and self.last_image_depth is not None:
             zval = self.last_image_depth.copy()
             if fill_value is not None:
                 zval[self._image_depth_null] = fill_value
```

Comment:
We can't just assume that exactly the null pixels are NaN, right?

Reply:
I think this is ok here.  The null pixels aren't NaN, they're just outside the camera clipping range and I think it's captured correctly with the _image_depth_null mask.